copy/copy.go

```diff
@@ -597,15 +597,32 @@ func (c *copier) createProgressBar(pool *mpb.Progress, info types.BlobInfo, kind
 		prefix = prefix[:maxPrefixLen]
 	}
 
-	bar := pool.AddBar(info.Size,
-		mpb.BarClearOnComplete(),
-		mpb.PrependDecorators(
-			decor.Name(prefix),
-		),
-		mpb.AppendDecorators(
-			decor.OnComplete(decor.CountersKibiByte("%.1f / %.1f"), " "+onComplete),
-		),
-	)
+	// If we don't know the size of the blob (size == -1), just display a
+	// spinner. Otherwise, display a proper progress bar.
+	var bar *mpb.Bar
+	if info.Size == -1 {
```

Comment:
Is there a guarantee that the only invalid size will be `-1`, and never zero or -2 or anything else?

And, minor point, I tend to like coding for the usual-case first (vs rare-case). What would you think of:
```go
    if info.Size > 0 {
        bar = pool.AddBar(...)
    } else {
        // Unknown or invalid size
        bar = pool.AddSpinner()
    }
```

Reply:
+1 to the `<0' vs `==-1' thought form @edsantiago

Reply:
The definition of `BlobInfo.Size` explicitly says `-1`.

(The flip side for having the usual case first is that one then has to explicitly track (or, as in the above suggestion, comment) what the unusual case is. In this case I don’t care either way at all, neither about the ordering nor about the condition)

Reply:
2 pro and 1 neutral, so I'll change it as proposed.